developers/transforms/dynamicppl/index.qmd

```diff
@@ -0,0 +1,399 @@
+```
+
+::: {.callout-warning}
+This is just for demonstration purposes!
+You shouldn't be directly setting variables in a linked `varinfo` like this unless you know for a fact that the value will be compatible with the constraints of the model.
+:::
+
+Now, we see that the constraint `m < x` is no longer satisfied.
+Hence, one might expect that if we try to evaluate the model using this `VarInfo`, we should obtain an error.
+Here, `evaluate!!` returns two things: the model's return value itself (which we defined above to be a `NamedTuple`), and the resulting `VarInfo` post-evaluation.
+
+```{julia}
+retval, ret_varinfo = DynamicPPL.evaluate!!(model, vi_linked, DefaultContext())
+getlogp(ret_varinfo)
+```
+
+But we don't get any errors!
+Indeed, we could even calculate the 'log probability density' for this evaluation.
+
+To understand this, we need to look at the actual value which was used during the model evaluation.
+We can glean this from the return value (or from the returned `VarInfo`, but the former is easier):
+
+```{julia}
+retval
+```
+
+We can see here that the model evaluation used the value of `m` that we provided, but the value of `x` was 'updated'.
+
+The reason for this is that internally in a model evaluation, we construct the transformation function from the internal to the model representation based on the *current* realizations in the model!
+That is, we take the `dist` in a `x ~ dist` expression _at model evaluation time_ and use that to construct the transformation, thus allowing it to change between model evaluations without invalidating the transformation.
+
+Knowing that the distribution of `x` depends on the value of `m`, we can now understand how the model representation of `x` got updated.
+The linked `VarInfo` does not store the model representation of `x`, but only its linked internal representation.
+So, what happened during the model evaluation above was that the linked internal representation of `x` – which was constructed using the _original_ value of `m` – was transformed back into a new model representation using a _different_ value of `m`.
+
+We can reproduce the 'new' value of `x` by performing these transformations manually:
+
+```{julia}
+# Generate a fresh linked VarInfo (without the new / 'corrupted' values)
+vi_linked = link(vi, model)
+# See the linked internal representations
+DynamicPPL.getindex_internal(vi_linked, vn_m), DynamicPPL.getindex_internal(vi_linked, vn_x)
+```
+
+Now we update the value of `m` like we did before:
+
+```{julia}
+vi_linked[vn_m] = vi_linked[vn_x] + 1
+vi_linked[vn_m]
+```
+
+When evaluating the model, the distribution of `x` is now changed, and so is the corresponding inverse bijector:
+
+```{julia}
+new_dist_x = truncated(Normal(); lower=vi_linked[vn_m])
+new_f_inv = DynamicPPL.invlink_transform(new_dist_x)
+```
+
+and if we apply this to the internal representation of `x`:
+
+```{julia}
+new_f_inv(DynamicPPL.getindex_internal(vi_linked, vn_x))
+```
```

Comment:
I feel like this could use a final sentence pointing out that the value is the same we saw before.

Also, I feel rude leaving behind such a friendly text that you've bonded with over three articles without saying goodbye.